kover-gradle-plugin/docs/index.md

```diff
@@ -36,11 +42,638 @@ Gradle plugin that measures the coverage for tests running on the JVM and genera
+
+----
+
+### Single module Kotlin multiplatform project
+Kover supports coverage measurement only for JVM targets. 
+Source code specified not in the common and not JVM source sets are ignored. 
+As well as coverage from non-JVM tests is also not taken into account.
+
+Otherwise, the use of the plugin is identical to the use in the [single module Kotlin JVM project](#single-module-kotlin-jvm-project).
+
+Please refer to the [example project](https://github.com/Kotlin/kotlinx-kover/tree/main/kover-gradle-plugin/examples/jvm/single-kmp).
+
+----
+
+### Multi-module Kotlin JVM project
+Please refer to the [example project](https://github.com/Kotlin/kotlinx-kover/tree/main/kover-gradle-plugin/examples/jvm/merged).
+
+#### Application of Kover plugin
+[Apply the plugin](#apply-kover-gradle-plugin-in-project) in apply in each module of your Gradle build.
+It is recommended to apply the Kover Plugin in the root module, even if there is no source code or tests there.
+You [do not need to specify](#application-in-submodule-with-plugins-dsl) the plugin version in all submodules.
+
+After applying Kover Gradle plugin, during the running tests, the classes are modified (instrumented) when loaded into the JVM which may lead to some performance degradation, or affect concurrent tests.
+In order to learn more about instrumentation, and it's configuring, please refer to the [corresponded section](#instrumentation).
+
+#### Merging reports from different modules
+In order for classes from all necessary modules to be included in the report, as well as tests from these modules to be performed, you need to configure report aggregation.
+
+For this:
+1. select the module that will collect information about all classes and from which the report generation will be launched.
+  let's call such a module **merging module**. We recommend using the root module (root Gradle project) as a **merging module**.
+2. add dependencies in the **merging module** to all other modules whose classes should be included in the merged report
+```kotlin
+dependencies {
+    kover(project(":moduleA"))
+    kover(project(":moduleB"))
+}
+```
+
+#### Generating reports
+Suppose if we chose the root module as the merging module, then
+
+- To generate JaCoCo-compatible XML report, call `:koverXmlReport` Gradle task
+- To generate HTML report, call `:koverHtmlReport` Gradle task
+- To validate verification rule specified in project in [verification block](#specify-verification-rules-for-total-variant), call `:koverVerify` Gradle task
+- To generate binary report in IC format, call `:koverBinaryReport` Gradle task
+- To print coverage to the log, call `:koverLog` Gradle task
+
+If a module with the path `:foo` was selected as the merging module, then you will need to call commands like `:foo:koverXmlReport`, etc.
+
+**To generate reports, call commands only by the full path, do not use short names, like `koverXmlReport`.**
```

Comment:
Doing so will just ask for a report in all available projects. Is that a problem? AFAIU, we should get one merged report and several individuals from that. This is unnecessary work, but not really a problem — unless reports generators conflict with each other. If they DO conflict, I would say this is a UX problem, they shouldn't do that.

Reply:
They don't conflict with each other. However, not all users are familiar with the features of Gradle, and how the two forms of calling tasks differ.

By calling tasks by a short name, users can get an unexpected result: tests will be run for all modules (projects) in the assembly, which can take a very long time.

This can happen, for example, in this case:
there are projects A, B, C, D in the build. The merged report is generated only on projects A, B, C. However, if user call the task in this way `gradlew koverHtmlReport` then, among others, tests are executed in the D project.